spring-integration-zookeeper/src/main/java/org/springframework/integration/zookeeper/config/LeaderInitiatorFactoryBean.java

```diff
@@ -72,10 +84,24 @@ public LeaderInitiatorFactoryBean setPath(String path) {
 	}
 
 	public LeaderInitiatorFactoryBean setRole(String role) {
-		this.candidate = new DefaultCandidate(UUID.randomUUID().toString(), role);
+		this.candidate = newCandidate(role);
+		Assert.state(this.candidate != null,
+				String.format("The %s requires a candidate; newCandidate(role) returned none.", getClass().getName()));
 		return this;
```

Comment:
You should use the override method that takes a supplied to avoid the `String.format()` call unless the assertion failed... `() -> String.format(...)`.